### PR TITLE
fix: initialisation of bugsnag in chrome extension

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Not initialize Bugsnag if the SDK is running on Chrome extension.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-common/src/utilities/checks.ts [hunk 1]
```diff
@@ -64,6 +64,13 @@ const isDefinedNotNullAndNotEmptyString = (value: any): boolean =>
  */
 const isTypeOfError = (obj: any): obj is Error => obj instanceof Error;
 
+/**
+ * Determines if the SDK is running inside a chrome extension
+ * @returns boolean
+ */
+const isSDKRunningInChromeExtension = (): boolean =>
```
Comment (bardisg) [inline packages/analytics-js-common/src/utilities/checks.ts:71]: This method is more suited for the capabilities detection utils, lets split in a new file please

Comment (bardisg) [reply packages/analytics-js-common/src/utilities/checks.ts:71]: dhecks are for value/type checks

Comment (MoumitaM) [reply packages/analytics-js-common/src/utilities/checks.ts]: Done

Comment (saikumarrs) [inline packages/analytics-js-common/src/utilities/checks.ts]: I'd make this a state entry like `appEnvironment`. Maybe in the future, we can also collect information like the framework used, etc.

Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/checks.ts]: CC @bardisg

File: packages/analytics-js-common/__tests__/utilities/detect.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { isSDKRunningInChromeExtension } from '../../src/utilities/detect';
+
+describe('Detect env/feature/capabilities', () => {
+  describe('isSDKRunningInChromeExtension', () => {
+    it('should return true if SDK is running inside chrome extension', () => {
+      (window as any).chrome = {
```
Comment (saikumarrs) [inline packages/analytics-js-common/__tests__/utilities/detect.test.ts:6]: Let's undo this change after the test is done, please.
Each test case should have a clean and known start.

File: packages/analytics-js-common/src/utilities/checks.ts (hunk 1 above)
Comment (bardisg) [reply packages/analytics-js-common/src/utilities/checks.ts]: Lets see if we need something like that as part of the JS metrics project

File: packages/analytics-js-common/__tests__/utilities/detect.test.ts (hunk 2 above)
Comment (MoumitaM) [reply packages/analytics-js-common/__tests__/utilities/detect.test.ts:10]: Done

**Review by saikumarrs — APPROVED** [review]
